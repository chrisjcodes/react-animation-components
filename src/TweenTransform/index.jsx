import React from 'react';
import { node, object, string } from 'prop-types';
import { Transition } from 'react-transition-group';

import { defaultAnimationProps, getInlineStyles } from '../utilities.js';

const TweenTransform = ({ children, start, finish, ...props }) => {
    const pos = {
        entering: start,
        entered: finish,
        exiting: finish,
        exited: start,
    };

    return (
        <Transition {...props}>
            {status => (
                <div
                    style={{
                        ...getInlineStyles(props),
                        transform: pos[status],
                        transitionProperty: 'transform',
                    }}
                >
                    {children}
                </div>
            )}
        </Transition>
    );
};

TweenTransform.propTypes = {
    children: node.isRequired,
    start: string,
    style: object,
    finish: string,
};

TweenTransform.defaultProps = {
    ...defaultAnimationProps,
    start: 'none',
    finish: 'none',
};

export default TweenTransform;
