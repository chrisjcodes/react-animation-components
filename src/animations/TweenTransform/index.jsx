import React from 'react';
import { node, object, string } from 'prop-types';
import { Transition } from 'react-transition-group';

import {
    defaultAnimationProps,
    getInlineStyles,
    getTimeoutValue,
} from 'utilities';

const TweenTransform = ({ children, enter, exit, ...props }) => {
    const pos = {
        entering: enter,
        entered: enter,
        exiting: exit,
        exited: exit,
    };

    return (
        <Transition timeout={getTimeoutValue(props)} {...props}>
            {status => (
                <div
                    style={{
                        ...getInlineStyles(props),
                        transform: pos[status],
                        transitionProperty: 'transform',
                    }}
                >
                    {children}
                </div>
            )}
        </Transition>
    );
};

TweenTransform.propTypes = {
    appear: bool,
    children: node.isRequired,
    delay: number,
    duration: number,
    exit: string,
    enter: string,
    style: object,
    timingFn: string,
};

TweenTransform.defaultProps = {
    ...defaultAnimationProps,
    enter: 'none',
    exit: 'none',
};

export default TweenTransform;
