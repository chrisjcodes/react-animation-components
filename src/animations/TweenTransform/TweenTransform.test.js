import React from 'react';
import TweenTransform from './index.jsx';
import renderer from 'react-test-renderer';

describe('TweenTransform', () => {
    test('renders', () => {
        const component = renderer.create(<TweenTransform />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('can accept custom styles', () => {
        const component = renderer.create(
            <TweenTransform style={{ background: 'black' }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDelay with delay prop', () => {
        const component = renderer.create(<TweenTransform delay={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDuration with duration prop', () => {
        const component = renderer.create(<TweenTransform duration={1000} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionTimingFunction with timingFn prop', () => {
        const component = renderer.create(
            <TweenTransform timingFn="ease-in-out" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets start', () => {
        const component = renderer.create(
            <TweenTransform start="translateX(10em)" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets finish', () => {
        const component = renderer.create(
            <TweenTransform finish="translateX(10em)" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
