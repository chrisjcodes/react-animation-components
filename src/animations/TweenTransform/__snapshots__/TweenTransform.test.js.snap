// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweenTransform can accept custom styles 1`] = `
<div
  style={
    Object {
      "background": "black",
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform renders 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform sets finish 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform sets start 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform sets transitionDelay with delay prop 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "1000ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform sets transitionDuration with duration prop 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "1000ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease",
    }
  }
/>
`;

exports[`TweenTransform sets transitionTimingFunction with timingFn prop 1`] = `
<div
  style={
    Object {
      "transform": "none",
      "transitionDelay": "0ms",
      "transitionDuration": "500ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease-in-out",
    }
  }
/>
`;
